src/ai/backend/cli/interaction.py

```diff
@@ -0,0 +1,116 @@
+import ipaddress
+import os
+
+
+def ask_host(prompt: str, default: str = "127.0.0.1") -> str:
+    try:
+        default = default.replace("localhost", "127.0.0.1")
+        _ = ipaddress.ip_address(default)
+    except ValueError:
+        raise ValueError("IP host must be given as 127.0.0.1")
+
+    while True:
+        user_reply = input(f"{prompt}(default: {default}): ")
```

Comment:
How about inserting a single space after the prompt so that the caller does not need to manually take care of it?